spec/unit/packagers/rpm_spec.rb

```diff
@@ -252,7 +251,7 @@ module Omnibus
 
       it 'uses the correct command' do
         expect(subject).to receive(:shellout!)
-          .with(/rpmbuild -bb --buildroot/)
+          .with(/rpmbuild --target x86_64 -bb --buildroot/)
         subject.create_rpm_file
       end
```

Comment:
I tried it in a slightly different fashion, though I could have done it
wrong as I am still learning rspec:

https://github.com/chef/omnibus/commit/610869efd8b54f4f6b1877bb12a9bb901737ec6b

On Wed, Sep 9, 2015 at 7:17 AM Seth Chisamore <notifications@github.com>
wrote:

> In spec/unit/packagers/rpm_spec.rb
> <https://github.com/chef/omnibus/pull/542#discussion_r39047463>:
>
> > @@ -252,7 +251,7 @@ module Omnibus
> >
> >        it 'uses the correct command' do
> >          expect(subject).to receive(:shellout!)
> > -          .with(/rpmbuild -bb --buildroot/)
> > +          .with(/rpmbuild --target x86_64 -bb --buildroot/)
> >          subject.create_rpm_file
> >        end
>
> Something like this should work for you:
>
> diff --git a/spec/unit/packagers/rpm_spec.rb b/spec/unit/packagers/rpm_spec.rb
> index a73de3a..8c2d2bd 100644--- a/spec/unit/packagers/rpm_spec.rb+++ b/spec/unit/packagers/rpm_spec.rb@@ -19,6 +19,7 @@ module Omnibus
>      let(:project_root) { File.join(tmp_path, 'project/root') }
>      let(:package_dir)  { File.join(tmp_path, 'package/dir') }
>      let(:staging_dir)  { File.join(tmp_path, 'staging/dir') }+    let(:architecture) { 'x86_64' }
>
>      before do
>        Config.project_root(project_root)@@ -32,7 +33,9 @@ module Omnibus
>        create_directory("#{staging_dir}/SOURCES")
>        create_directory("#{staging_dir}/SPECS")
> -      stub_ohai(platform: 'redhat', version: '6.5')+      stub_ohai(platform: 'redhat', version: '6.5') do |data|+        data['kernel']['machine'] = architecture+      end
>      end
>
>      describe '#signing_passphrase' do@@ -251,7 +254,7 @@ module Omnibus
>
>        it 'uses the correct command' do
>          expect(subject).to receive(:shellout!)-          .with(/rpmbuild --target x86_64 -bb --buildroot/)+          .with(/rpmbuild --target #{architecture} -bb --buildroot/)
>          subject.create_rpm_file
>        end
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/omnibus/pull/542/files#r39047463>.
>